### ISSUE TITLE
device_mouse_raw_x not implemented

### ISSUE THREAD

**Issue by briviere** [enigma-dev/enigma-dev#1038]
It appear that device_mouse_raw_x has not been implemented? Unless it is in a newer branch?

thanks

Brian

**Comment by JoshDreamland** [comment]
I have never heard of that function. My guess is that Yo-yo renamed `display_mouse_get_x();`.

**Comment by faissaloo** [comment]
@JoshDreamland https://docs.yoyogames.com/source/dadiospice/002_reference/mouse,%20keyboard%20and%20other%20controls/device%20input/device_mouse_raw_x.html  
>This function returns raw x position of the device. What this means is that it returns the actual device definition of the x position that is being touched, not the GameMaker: Studio one, and as such will ignore things like view position and scaling.

**Comment by JoshDreamland** [comment]
That description seems to match `display_mouse_get_x()` perfectly. The actual difference appears to be the extra int parameter, for multitouch support, I presume. Multitouch isn't terribly relevant to PC ENIGMA, so I'd suggest just using the `display` equivalent. If it still takes window coordinates into account, there's also `window_mouse_get_y`.

**Comment by RobertBColton** [comment]
Originally I thought when they say "raw" input they are literally referring to WM_INPUT aka RAWINPUT multiple device handling. The differences are not just in the API, but also in the way the mouse input will be received.
https://docs.microsoft.com/en-us/windows/win32/inputdev/about-raw-input

However, it does not seem to be the case as several people mention that it only works with touch, and does not work with multiple mice correctly.
https://www.reddit.com/r/gamemaker/comments/berk4b/is_there_any_way_to_get_input_from_multiple_mice/

**Comment by RobertBColton** [comment]
Yeah I think this is just touch support with multiple fingers, possibly using the following SDL API. The "raw" in the GMS function name just means it's not offset by the views and such like `device_mouse_x` is.
https://wiki.libsdl.org/SDL_TouchFingerEvent
